src/components/List/DescriptionList.jsx

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright (C) 2018 The Trustees of Indiana University
+SPDX-License-Identifier: BSD-3-Clause
+*/
+import classNames from "classnames";
+import * as React from "react";
+import * as PropTypes from "prop-types";
+
+import * as Rivet from "../util/Rivet";
+
+const classPrefix = "rvt-list-description";
+
+/**
+ * Use the decription list show a set of terms and descriptions.
+ */
+const DescriptionList = ({
+  id = Rivet.shortuid(),
+  className,
+  children,
+  ...attrs
+}) => {
+  return (
+    <dl className={classNames(classPrefix, className)} id={id} {...attrs}>
+      {asListItems(children)}
+    </dl>
+  );
+};
+DescriptionList.displayName = "DescriptionList";
+DescriptionList.propTypes = {
+  /** A unique identifier for the description list */
+  id: PropTypes.string,
+};
+
+const asListItems = (children) =>
+  children
+    ? React.Children.map(children, (child, index) => {
+        if (
+          typeof child === "string" ||
+          typeof child === "number" ||
+          !["dd", "dt"].includes(child.type)
+        ) {
+          return index % 2 === 0 ? <dt>{child}</dt> : <dd>{child}</dd>;
+        }
+        return child;
```

Comment:
In regards to not using the Children API:

1. In case where client sends `<dt>` or `<dd>` elements inside the component, you can just return `children` as it is.
2. If they want to send an array, you could do something similar to [this pattern](https://beta.reactjs.org/reference/react/Children#accepting-an-array-of-objects-as-a-prop).

Reply:
I don't think that inspecting or modifying the children is needed in the DescriptionList or List components. Especially for `dl`, there are lots of valid arrangements that makes managing it likely more difficult than it is worth. For example, you can have multiple `dd` following a `dt`, not just zero or one. `dt` and `dd` elements can be grouped with `div`, meaning a `div` would be the direct child of the `dl`.

If the user wants to use an array or object to generate the children, that's fine. But it is a user concern, not a library concern. For example:

```jsx
<!-- Not this -->
<List>{["Item 1", "Item 2", "Item 3"]}</List>

<!-- Recommended approach -->
<List>
  {["Item 1", "Item 2", "Item 3"].map((item) => (<li>{item}</li>)}
</List>
```